src/storage/test/QueryBoundBenchmark.cpp

```diff
@@ -0,0 +1,186 @@
+    for (auto tagId = 3001; tagId < 3010; tagId++) {
+        schema->addTagSchema(
+            0 /*space id*/, tagId, TestUtils::genTagSchemaProvider(tagId, 3, 3));
+    }
+    mockData(gKV.get());
+}
+
+cpp2::GetNeighborsRequest buildRequest(bool outBound = true) {
+    cpp2::GetNeighborsRequest req;
+    req.set_space_id(0);
+    decltype(req.parts) tmpIds;
+    for (auto partId = 0; partId < FLAGS_req_parts; partId++) {
+        for (auto vertexId = 1; vertexId <= FLAGS_vrpp; vertexId++) {
+            tmpIds[partId].push_back(vertexId);
+        }
+    }
+    req.set_parts(std::move(tmpIds));
+    req.set_edge_type(outBound ? 101 : -101);
+    // Return tag props col_0, col_2, col_4
+    decltype(req.return_columns) tmpColumns;
+    for (int i = 0; i < 3; i++) {
+        tmpColumns.emplace_back(
+            TestUtils::propDef(cpp2::PropOwner::SOURCE,
+                               folly::stringPrintf("tag_%d_col_%d", 3001 + i*2, i*2),
+                               3001 + i*2));
+    }
+    tmpColumns.emplace_back(TestUtils::propDef(cpp2::PropOwner::EDGE,
+                                               folly::stringPrintf("_dst")));
+    tmpColumns.emplace_back(TestUtils::propDef(cpp2::PropOwner::EDGE,
+                                               folly::stringPrintf("_rank")));
+    // Return edge props col_0, col_2, col_4 ... col_18
+    for (int i = 0; i < 10; i++) {
+        tmpColumns.emplace_back(
+            TestUtils::propDef(cpp2::PropOwner::EDGE,
+                               folly::stringPrintf("col_%d", i*2)));
+    }
+    req.set_return_columns(std::move(tmpColumns));
+    return req;
+}
+
+}  // namespace storage
+}  // namespace nebula
+
+void run(int32_t iters, int32_t handlerNum) {
+    FLAGS_max_handlers_per_req = handlerNum;
+    auto req = nebula::storage::buildRequest();
+    auto executor = std::make_unique<folly::CPUThreadPoolExecutor>(FLAGS_handler_num);
+    for (decltype(iters) i = 0; i < iters; i++) {
+        auto* processor
+                = nebula::storage::QueryBoundProcessor::instance(
+                                                            gKV.get(),
+                                                            schema.get(),
+                                                            executor.get(),
+                                                            nebula::storage::BoundType::OUT_BOUND);
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+    }
+}
+
+BENCHMARK(query_bound_1, iters) {
+    run(iters, 1);
+}
+
+BENCHMARK(query_bound_3, iters) {
+    run(iters, 3);
+}
+
+BENCHMARK(query_bound_10, iters) {
+    run(iters, 10);
+}
+/*************************
+ * End of benchmarks
+ ************************/
+
+
+int main(int argc, char** argv) {
+    folly::init(&argc, &argv, true);
+    nebula::fs::TempDir rootPath("/tmp/QueryBoundBenchmarkTest.XXXXXX");
```

Comment:
I'm not sure this benchmark need to IO (I can't read it from context). if it is, HDD or SSD matters.

Reply:
It needs to read data from rocksdb.  And the benchmark is based on SSD.

Reply:
By the way, as the test data is not large enough,  all requests would hit the cache.

Reply:
a better way is to warm cache in a prepare step. and then do the benchmarks.

Reply:
We just test the same request with single thread VS multi threads,  so no really needs with warming step.